tests/ert_tests/ert3/data/test_transmitters.py

```diff
@@ -139,9 +189,29 @@ async def test_simple_record_transmit_and_load(
     ) as record_transmitter_factory:
         transmitter = record_transmitter_factory(name="some_name")
         await transmitter.transmit_record(record_in)
-
         record = await transmitter.load()
-        assert record.data == expected_data
+        if isinstance(record_in, NumericalRecordTree):
+            sub_records = json.loads(record.data.decode("utf-8"))
+            transmitters = await transmitter.get_recordtree_transmitters(
+                sub_records, RecordType.MAPPING_STR_FLOAT
+            )
+            record_tree = await transmitter.load_numerical_recordtree(transmitters)
+            flat_data = {
+                _path: _rec.data for _path, _rec in record_tree.flat_record_dict.items()
+            }
+            assert _transmitter._unflatten_record_dict(flat_data) == expected_data
+        elif isinstance(record_in, BlobRecordTree):
+            sub_records = json.loads(record.data.decode("utf-8"))
+            transmitters = await transmitter.get_recordtree_transmitters(
+                sub_records, RecordType.BYTES
+            )
+            record_tree = await transmitter.load_blob_recordtree(transmitters)
+            flat_data = {
+                _path: _rec.data for _path, _rec in record_tree.flat_record_dict.items()
+            }
+            assert _transmitter._unflatten_record_dict(flat_data) == expected_data
+        else:
+            assert record.data == expected_data
```

Comment:
I think it would be much nicer API for the transmitter if all of this was treated as an implementation detail. I would expect as a user/developer that the following API would work also for the record tree: 

```python
await transmitter.transmit_record(record_in)
record = await transmitter.load()
```
That fact that these are sub-transmitters (or something else) I think should be hidden in the implementation.

Reply:
I fully agree, there are two small issues though: 
1) we might not need to load the entire `RecordTree` or what does it mean to `load` record tree per say.
2) currently we don't know that the BlobRecord (containing the metadata for subrecords) is actually a RecordTree

Reply:
1. I don't think we need to handle partial loading at this point in time, but I think it could be done by having a optional argument `path` to load that subtree. 
2. If I understand you correctly, this means that when the Transmitter loads the record tree, it does not have enough information to tell if it should load a Blob record or a record tree? If that is the case I think we should extend the transmitter to have knowledge of this, in the same way is it knows that it should load a numerical or blob record.